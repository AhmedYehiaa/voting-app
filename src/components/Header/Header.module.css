.header {
  z-index: 1;
  width: 100%;
  background: rgb(0, 206, 165);
}

.logo {
  max-width: 90%;
  margin: auto;
  display: flex;
  align-items: center;
  font-family: "Solway", serif;
}

.header img {
  max-height: 50px;
}
.header p {
  font-weight: bold;
}
