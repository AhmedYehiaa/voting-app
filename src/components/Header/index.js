import React from "react";
import logo from "../../assets/vote.svg";
import styles from "./Header.module.css";

const Header = () => {
  return (
    <header className={styles.header}>
      <div className={styles.logo}>
        <img src={logo} alt="logo" />
        <p>Create simple polls quickly and easily</p>
      </div>
    </header>
  );
};

export default Header;
