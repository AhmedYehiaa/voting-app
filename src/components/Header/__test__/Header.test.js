import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Header from "../index";

afterEach(cleanup);

describe("Header Component", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Header />, div);
  });

  it("renders Header with text ", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("header")).toHaveTextContent(
      "Create simple polls quickly and easily"
    );
  });

  it("renders Header with img ", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("img")).toBeEnabled();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
