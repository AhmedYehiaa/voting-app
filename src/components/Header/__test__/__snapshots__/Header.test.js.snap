// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<header
  className="header"
  data-testid="header"
>
  <div
    className="logo"
  >
    <img
      alt="logo"
      data-testid="img"
      src="vote.svg"
    />
    <p>
      Create simple polls quickly and easily
    </p>
  </div>
</header>
`;
